src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -262,8 +265,228 @@ void emberAfOnOffClusterServerInitCallback(EndpointId endpoint)
+    }
+
+    // Only apply effect if OnOff is on
+    if (isOnBeforeCommand)
+    {
+        OnOffEffect * effect = inst(endpoint);
+
+        if (effect != nullptr && effect->mOffWithEffectTrigger != nullptr)
+        {
+            effect->mEffectIdentifier = effectId;
+            effect->mEffectVariant    = effectVariant;
+
+            effect->mOffWithEffectTrigger(effect);
+        }
+    }
+
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+bool OnOffServer::OnWithRecallGlobalSceneCommand()
+{
+    chip::EndpointId endpoint = emberAfCurrentEndpoint();
+    EmberAfStatus status      = EMBER_ZCL_STATUS_SUCCESS;
+
+    bool globalSceneControl = false;
+    OnOff::Attributes::GlobalSceneControl::Get(endpoint, &globalSceneControl);
+
+    if (globalSceneControl)
+    {
+        emberAfSendImmediateDefaultResponse(status);
+        return true;
+    }
+
+    OnOff::Attributes::GlobalSceneControl::Set(endpoint, true);
+    setOnOffValue(endpoint, Commands::On::Id, false);
+
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+bool OnOffServer::OnWithTimedOffCommand(BitFlags<OnOffControl> onOffControl, uint16_t onTime, uint16_t offWaitTime)
+{
+    EmberAfStatus status      = EMBER_ZCL_STATUS_SUCCESS;
+    chip::EndpointId endpoint = emberAfCurrentEndpoint();
+
+    bool isOn = false;
+    OnOff::Attributes::OnOff::Get(endpoint, &isOn);
+
+    // OnOff is off and the commands is only accepted if on
+    if (onOffControl.Has(OnOffControl::kAcceptOnlyWhenOn) && !isOn)
+    {
+        emberAfSendImmediateDefaultResponse(status);
+        return true;
+    }
+
+    uint16_t currentOffWaitTime = MAX_TIME_VALUE;
+    OnOff::Attributes::OffWaitTime::Get(endpoint, &currentOffWaitTime);
+
+    uint16_t currentOnTime = 0;
+    OnOff::Attributes::OnTime::Get(endpoint, &currentOnTime);
+
+    if (currentOffWaitTime > 0 && !isOn)
+    {
+        uint16_t newOffWaitTime = currentOffWaitTime < offWaitTime ? currentOffWaitTime : offWaitTime;
+        OnOff::Attributes::OffWaitTime::Set(endpoint, newOffWaitTime);
+
+        currentOffWaitTime = newOffWaitTime;
+    }
+    else
+    {
+        uint16_t newOnTime = currentOnTime > onTime ? currentOnTime : onTime;
+        OnOff::Attributes::OnTime::Set(endpoint, newOnTime);
+
+        OnOff::Attributes::OffWaitTime::Set(endpoint, offWaitTime);
+        setOnOffValue(endpoint, Commands::On::Id, false);
+
+        currentOnTime      = newOnTime;
+        currentOffWaitTime = offWaitTime;
+    }
+
+    emberAfOnOffClusterPrintln("On Time:  %d | off wait Time: %d", currentOnTime, currentOffWaitTime);
+
+    if (currentOnTime < MAX_TIME_VALUE && currentOffWaitTime < MAX_TIME_VALUE)
+    {
+        emberEventControlSetDelayMS(configureEventControl(endpoint), UPDATE_TIME_MS);
+    }
+
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+/**
+ * @brief Updates OnOff values after timer is finished
+ *
+ * @param endpoint
```

Comment:
consider cleaning up some of these doccomments. the `@param endpoint` does not help me reading code - I can see this param in the method declaration.

Reply:
done